src/helpers/clipboard/clipboard_cells_state.ts

```diff
@@ -108,6 +102,20 @@ export class ClipboardCellsState extends ClipboardCellsAbstractState {
   }
 
   isPasteAllowed(target: Zone[], clipboardOption?: ClipboardOptions): CommandResult {
+    const updatedCells: ClipboardCell[][] = [];
+    for (const cell of this.cells) {
+      const rowCells: ClipboardCell[] = [];
+      for (const rowCell of cell) {
+        rowCells.push(this.getCellData(rowCell.position));
+      }
+      updatedCells.push(rowCells);
+    }
+
+    if (!deepEquals(this.cells, updatedCells) && this.operation === "CUT") {
+      // cannot paste if content or style in cut zone is altered before paste
+      this.cells = [];
```

Comment:
the state should only be modified by commands (see command query separation pattern [1])
Here the state is modified even though the command is being rejected.
Besides, rejecting the command is not what we want. We don't want to prevent updating a cell in the clipped zones.

The clipboard state should be invalidated in the clipboard plugin, by handling `UPDATE_CELL` commands.
You can add an abstract method `isInvalidatedBy(cmd: Command): boolean`,  handle `UPDATE_CELL` commands in there and return `true`/`false` depending on the clipped zones.
In a following commit, you can remove `isColRowDirtyingClipboard` and move the implementation in the new `isInvalidatedBy` by handling `ADD_COLUMNS_ROWS` and `REMOVE_COLUMNS_ROWS`

[1] https://github.com/odoo/o-spreadsheet/blob/saas-16.3/doc/extending/architecture.md#model-commands-and-getters

Reply:
Hi! Sorry for the delayed response.
I understood what you wanted to imply and changed my approach accordingly. It took a while since I faced some edge cases while solving the issue.
Although I've implemented the necessary changes now. Thank you!